handout/modules/introduction.md

```diff
@@ -0,0 +1,89 @@
+# What is an Angular 2 Module?
+
+In Angular 2, a module is a mechanism to group components, directives, pipes and services that are related, in such a way that can be combined with other modules to create an application. An Angular 2 application can be thought of as a puzzle where each piece (or each module) is needed to be able to see the full picture.
+
+Another analogy to understand Angular 2 modules is classes. In a class, we can define public or private methods. The public methods are the API that other parts of our code can use to interact with it while the private methods are implementation details that are hidden. In the same way, a module can export or hide components, directives, pipes and services. The exported elements are meant to be used by other modules, while the ones that are not exported (hidden) are just used inside the module itself and cannot be directly accessed by other modules of our application.
+
+## A Basic Use of Modules
+
+To be able to define modules we have to use the decorator `NgModule`.
+
```

Comment:
@smithad15 I didn't put the filename there because that snippet was mostly to give a an overview of how the `NgModule` decorator looks like. I'm "formally" starting the code example in the next snippet and that's when I put the filename so people can code along if they want to.